packages/core-sdk/resolvers/token-resolvers/resolvers/uniswap/src/query/functions/isValidProtocolToken.ts

```diff
@@ -0,0 +1,44 @@
+import { pairAddress } from "../utils/addressUtils";
+import {
+  env,
+  Ethereum_Connection,
+  Ethereum_Query,
+  Input_isValidProtocolToken,
+  QueryEnv,
+} from "../w3";
+
+function isValidUniswapV2Pool(tokenAddress: string, connection: Ethereum_Connection): boolean {
```

Comment:
I would use something like a registry to validate this, here we are just using the input to validate the input. 
poolAddress is the input in this case which can be any arbitrary contract. There can be a malicious contract using the same algorithm to calculate pool address and deploy the contract to it and pretend to be uniswap pool contract. Without verifying it through Uniswap registry, we can't be sure here. 

This is just the theory and may not even be possible or very unlikely to happen in practice but Ideally, I wouldn't want to validate input using just the input. Let me know what do you think

Reply:
I hear you, and I know it seems at first glance like it would be insecure. But it's more secure than you might think!

It's secure because the algorithm used to calculate the pool address uses a hash of the pool contract's code. If someone wanted to launch a malicious contract at the same address, the contract would have to be a Pool contract and therefore could not be malicious. The algorithm mimics the `create2` EVM opcode that the EVM uses to assign contract addresses. Uniswap and Sushiswap use this algorithm in their JavaScript SDK's because it's more performant than querying the chain.

From [EIP-1014](https://eips.ethereum.org/EIPS/eip-1014):
> Allows interactions to (actually or counterfactually in channels) be made with addresses that do not exist yet on-chain but can be relied on to only possibly eventually contain code that has been created by a particular piece of init code. Important for state-channel use cases that involve counterfactual interactions with contracts.

Reply:
But also it would be really easy to use the registry, and it would simplify the code. I'm fine with doing that. :)

Reply:
Oh, I didn't know this but If this is the case then this looks good to me. Thanks for letting me know.